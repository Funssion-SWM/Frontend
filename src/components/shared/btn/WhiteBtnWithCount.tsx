type Props = {
  text: string;
  count?: number;
  onClickBtn: () => void;
  onClickCount: () => void;
  extraStyle?: string;
};

export default function WhiteBtnWithCount({
  text,
  count = 0,
  onClickBtn,
  onClickCount,
  extraStyle = '',
}: Props) {
  return (
    <span className={extraStyle}>
      <button
        type="button"
        className={`bg-white text-soma-blue-50 border-soma-grey-40 border-r-0 border-[1px] pl-3.5 pr-1 py-2 rounded-l-3xl transition hover:bg-soma-grey-40 text-xs sm:text-sm`}
        onClick={() => onClickBtn()}
      >
        {text}
      </button>
      <button
        type="button"
        className={`bg-white text-soma-blue-50 border-soma-grey-40 border-l-0 border-[1px] pr-3.5 pl-1 py-2 rounded-r-3xl transition hover:bg-soma-grey-40 text-xs sm:text-sm`}
        onClick={() => onClickCount()}
      >
        {count}
      </button>
    </span>
  );
}
