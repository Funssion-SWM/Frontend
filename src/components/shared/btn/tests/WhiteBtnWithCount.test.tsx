import { render, screen } from '@testing-library/react';
import WhiteBtnWithCount from '../WhiteBtnWithCount';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

describe('WhiteBtnWithCount', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <WhiteBtnWithCount
        text="text"
        count={0}
        onClickBtn={() => {}}
        onClickCount={() => {}}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button or count is clicked', async () => {
    const mockBtnFunction = jest.fn();
    const mockCountFunction = jest.fn();

    render(
      <WhiteBtnWithCount
        text="text"
        count={0}
        onClickBtn={mockBtnFunction}
        onClickCount={mockCountFunction}
      />
    );

    const [btn, count] = screen.getAllByRole('button');

    await userEvent.click(btn);
    expect(mockBtnFunction).toHaveBeenCalledTimes(1);

    await userEvent.click(count);
    expect(mockCountFunction).toHaveBeenCalledTimes(1);
  });
});
