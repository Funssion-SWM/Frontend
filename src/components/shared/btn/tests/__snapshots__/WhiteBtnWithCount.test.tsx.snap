// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhiteBtnWithCount renders correctly 1`] = `
<div
  className=""
>
  <button
    className="bg-white text-soma-blue-50 border-soma-grey-40 border-r-0 border-[1px] pl-3.5 pr-1 py-2 rounded-l-3xl transition hover:bg-soma-grey-40 text-xs sm:text-sm"
    onClick={[Function]}
    type="button"
  >
    text
  </button>
  <button
    className="bg-white text-soma-blue-50 border-soma-grey-40 border-l-0 border-[1px] pr-3.5 pl-1 py-2 rounded-r-3xl transition hover:bg-soma-grey-40 text-xs sm:text-sm"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
</div>
`;
